crypto/dilithium/pqcrystals_dilithium_ref_common/README.md

```diff
@@ -0,0 +1,12 @@
+# AWS-LC ML-DSA readme file
+
+The source code in this folder implements ML-DSA as defined in FIPS 204 Module-Lattice-Based Digital Signature Standard [link](https://csrc.nist.gov/pubs/fips/204/final).
+
+**Source code origin and modifications.** The source code was imported from a branch of the official repository of the Crystals-Dilithium team: https://github.com/pq-crystals/dilithium. The code was taken at [commit](https://github.com/pq-crystals/dilithium/commit/cbcd8753a43402885c90343cd6335fb54712cda1) as of 10/01/2024. At the moment, only the reference C implementation is imported.
+
+The `api.h`, `fips202.h` and `params.h` header files were modified to support our [prefixed symbols build](https://github.com/awslabs/aws-lc/blob/main/BUILDING.md#building-with-prefixed-symbols).
```

Comment:
It may be helpful to describe how they were modified in order to help future maintainers that have to deal with conflicts.

Reply:
I completely agree. There are going to be further changes to the Dilithium code, then I will update this readme after to reflect all changes. I've left the note about the prefix build stuff in here as-is for now, as in a following PR I will be removing all the prefix build stuff for Dilithium. Essentially, the same changes made to ML-KEM (https://github.com/aws/aws-lc/commit/7b5ba898c012bb96a68b31553912471f7f2653df) will be made for ML-DSA. Once those changes are in, I will update this text with similar text from the [ML-KEM readme](https://github.com/aws/aws-lc/tree/main/crypto/fipsmodule/ml_kem#readme).

Reply:
Addressed in [72d7271](https://github.com/aws/aws-lc/pull/1894/commits/72d7271fbfcb893a09fc508c91ee25846724eb92). Will continue to update this readme with additional changes.

Reply:
Added more detail in [7f31525](https://github.com/aws/aws-lc/pull/1894/commits/7f3152579fc55008fc484df527e205f4954028dd).